src/librustdoc/passes/calculate_doc_coverage.rs

```diff
@@ -41,10 +41,13 @@ impl ItemCount {
         has_docs: bool,
         has_doc_example: bool,
         should_have_doc_examples: bool,
+        should_have_docs: bool,
     ) {
-        self.total += 1;
+        if should_have_docs {
```

Comment:
This should use the same logic as doc examples, which counts it _either_ if it has docs or should have docs. That way, rustdoc doesn't mark your crate as having a lower percentage documented than it actually has.
```suggestion
        if should_have_docs ||  has_docs {
```